### ISSUE TITLE
Parameter must be bound before case context

### ISSUE THREAD

**Issue by ahirner** [PrefectHQ/prefect#2604]
## Current behavior 0.11.0
`case.__enter__` mimics python if blocks, yet assumes that parameters are bound to the flow outside its context. That can cause some head-scratching for users.
```python
from prefect import task, Parameter, Flow
from prefect.tasks.control_flow import merge, case

with Flow("test maybe first param use") as f:
    x = Parameter("x")
    p = Parameter("p")
    with case(p > 10, True):
        y = x-p
    y = merge(y, p)

state = f.run(parameters=dict(x=0, p=11))
assert state.result[y].result == -11
state = f.run(parameters=dict(x=0, p=9))
assert state.result[y].result == 9
```
**Errors** with:
```
Traceback (most recent call last):
  File "pref_case_bind.py", line 8, in <module>
    y = x-p
  File "~/miniconda3/lib/python3.6/site-packages/prefect/tasks/control_flow/case.py", line 100, in __exit__
    child.set_upstream(cond, flow=self._flow)
  File "~/miniconda3/lib/python3.6/site-packages/prefect/core/task.py", line 591, in set_upstream
    self.set_dependencies(flow=flow, upstream_tasks=[task], mapped=mapped)
  File "~/miniconda3/lib/python3.6/site-packages/prefect/core/task.py", line 566, in set_dependencies
    mapped=mapped,
  File "~/miniconda3/lib/python3.6/site-packages/prefect/core/flow.py", line 811, in set_dependencies
    mapped=is_mapped,
  File "~/miniconda3/lib/python3.6/site-packages/prefect/core/flow.py", line 482, in add_edge
    "Parameters must be root tasks and can not have upstream dependencies."
ValueError: Parameters must be root tasks and can not have upstream dependencies.
```
**Passes** if: `x = Parameter("x")()`

## Proposed behavior
If `child` is a `Parameter` we would `.bind(self_flow)` between L#99 and L#100. I cannot think of any unintended side-effects, because use outside of a case context also binds yet unbound parameters (please confirm though).

## Example
See above

**Comment by jcrist** [comment]
Thanks for the excellent issue report, this should be fixed by #2608.

**Comment by ahirner** [comment]
It does fix also our real flows, thx!
